### ISSUE TITLE
Frost Nova interrupt channeled spells. (BB #473)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#473]
This issue was migrated from bitbucket.
**Original Reporter:** Alex_Step
**Original Date:** 07.01.2014 11:21:26 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/473

<hr>

Mage spell Frost Nova - should not interrupt cast any spells.. Now this spell interrupt channeled spells, for example: Evocation, Arcane Missles, Drain Mana

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Yooki
**Original Date:** 07.01.2014 14:50:34 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/473/#comment-7621238

<hr>

Did you update your core with the newest revision available? There was a recent repository update that should have fixed that.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Alex_Step
**Original Date:** 07.01.2014 16:20:23 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/473/#comment-7622967

<hr>

You're talking about?https://bitbucket.org/oregon/oregoncore/commits/d323daff8a25671413d6abe7ad5b1379913b86d6 

This no fixed Frost Nova bug..

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Yooki
**Original Date:** 07.01.2014 16:38:29 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/473/#comment-7623195

<hr>

My apologies, I was not aware my pull request has not been approved yet. 
The update I was referring to is laying dormant in the pull request section: 
https://bitbucket.org/oregon/oregoncore/pull-request/51/resolved-issue-108-spell-channeling-and 

Pull the updates from my forked repository, patch your core and let me know if the issue persists. Hopefully that update fixes all problems related to spell channel interruption.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 07.01.2014 19:23:37 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/473/#comment-7625469

<hr>

Do you have proof that it should not interrupt spell casting?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Alex_Step
**Original Date:** 07.01.2014 20:22:41 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/473/#comment-7626094

<hr>

For example wowwiki:
Use Frost Nova to root enemies in place and then gain some range to cast spells without getting hit. This is especially effective against melee-only enemies, as they will not have any other way to attack you. Against caster-type enemies, use Frost Nova to gain range along with [Counterspell] to interrupt their casting.

If it were possible interrupt, why mention about [Counterspell]?)
Casted normal spells - not interrupted Frost Nova, only Channeled.. it is not logical..
For example Channeled spells not interrupt Fireball etc.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Yooki
**Original Date:** 07.01.2014 20:32:08 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/473/#comment-7626193

<hr>

Alex, did you update your core with the patch I've linked in my previous post?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Alex_Step
**Original Date:** 07.01.2014 20:41:02 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/473/#comment-7626306

<hr>

Yes update, no fixed this..

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 07.01.2014 20:47:57 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/473/#comment-7626394

<hr>

OH
It interrupts other players, not your own.
Makes sense now.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 09.01.2014 02:43:40 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/473/#comment-7647052

<hr>

This is a much larger problem then it seems, any spell that roots a target in place causes channels to cast.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 10.01.2014 03:03:35 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/473/#comment-7664850

<hr>

Resolved, Dirty please close

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Alex_Step
**Original Date:** 10.01.2014 08:53:20 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/473/#comment-7667391

<hr>

Tested, problem solved.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** release_ok
**Original Date:** 10.01.2014 09:07:18 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/473/#comment-7667535

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** release_ok
**Original Date:** 11.01.2014 08:57:15 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/473/#comment-7680631

<hr>

Resolved issue #473 - Channeled spells interrupt:

This patch contains some small changes that determine when a channeled spell should be interrupted.
Here are the things that have been 'fixed' or improved with this commit:
- Channeled spells will no longer be interrupted when you re-cast the spell before the channel
  timer has expired. This action would cause a visual bug that would as Usagi nicely put it,
  "Be an ugly reminder that you were on a private server".
- Fixed an issue that caused yet another visual bug. This issue is very similar to the problem
  mentioned above, with the only difference being it affects npcs and not players.
  This would be the official description:
  
  "Say a creature was casting a channeled spell, and you as a warrior or something other
  interrupted that channel. It would still show that the creature is channeling to the client"
- It would seem some channeled spells would get interrupted if the caster was rooted before
  finishing the cast. I personally have not experienced this issue, but I am being assured this
  is a valid and existing problem. Root spells had an attribute appended to them by the core that
  caused this issue. The attribute was related to taking damage if I am not mistaken, but I can't be sure.
  
  And so it should be safe to say that the issue #473 (Frost Nova interrupt channeled spells)
  has been resolved with this change, as well as other similar issues with other spells not mentioned.
- Implemented a new condition every channeled spell has to fulfill if they do not want to be interrupted.
  Spells can no longer be channeled if the caster is not facing the target at all times. If the target
  exits the casters front visual 120 degree arc, the spell will be interrupted.
  
  This is not something that is 'Blizzlike', but does make sense and does not break immersion.
  And this could add more challenge to the game, since now it's slightly more difficult to focus channels.
- Repaired my earlier commit (4e4f88f) that fixed the 'shadowmeld in combat issue'.
  The problem reappeared on a few occasions so I had to look into this, and make yet another 'fix for a fix'.
  
  Bunch of hugs and thanks to LordUsagi for crafting the first three additions to this commit.

→ <<cset 3d318beb9b75>>
